lib/components/form/SiteDetails.tsx

```diff
@@ -57,12 +76,30 @@ const SiteDetails: FC<SiteDetailsProps> = ({ site }) => {
     });
   }
 
-  const { trigger } = useSWRMutation(
+  const { trigger: triggerPOST } = useSWRMutation(
     `${process.env.NEXT_PUBLIC_API_BASE_URL_POST}/sites`,
-    sendRequest
+    sendPostRequest
+  );
+
+  async function sendPutRequest(url: string, { arg }: { arg: FormPostData }) {
+    const options = await getAuthenticatedRequestOptions(url);
+    return fetch(url, {
+      method: 'PUT',
+      body: JSON.stringify(arg),
+      credentials: 'include',
+      headers: {
+        'Content-Type': 'application/json',
+        ...options.headers,
+      },
+    });
+  }
```

Comment:
`sendPutRequest` and `sendPostRequest` are pretty similar so we could try to avoid code duplication like this 
```
async function sendRequest(method: string, url: string, { arg }: { arg: FormPostData }) {
  const options = await getAuthenticatedRequestOptions(url);
  return fetch(url, {
    method: method,
    body: JSON.stringify(arg),
    credentials: 'include',
    headers: {
      'Content-Type': 'application/json',
      ...options.headers,
    },
  });
}
```